### PR TITLE
Normalize heredoc delimiters

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
# Description:

On another PR, I noticed that while normally we use `RUBY` as the delimiter for heredocs including ruby code, sometimes we use `RB`.

`RUBY` is better so I'm replacing `RB` occurrences with `RUBY`.

# Tasks:

- [x] Describe the problem / feature
- [ ] Write tests
- [x] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by simi** [comment]
:thinking: I see there are various `HEREDOC` names used across the code. Some are lower case, some upper case, some have meaningful name like `RUBY`, some have less meaningful name like `D` or `EOF`.

There are two related cops we can consider to use as well if you're doing `HEREDOC` cleanup.

https://rubocop.readthedocs.io/en/latest/cops_naming/#namingheredocdelimitercase
https://rubocop.readthedocs.io/en/latest/cops_naming/#namingheredocdelimiternaming

**Comment by deivid-rodriguez** [comment]
Good point @simi. I enabled `Naming/HeredocLimiterNaming` to check for the specific issue I wanted to normalize, and also enabled the delimiter case cop.

**Comment by deivid-rodriguez** [comment]
@simi You ok with this?

File: lib/rubygems.rb [hunk 1]
```diff
@@ -401,11 +401,11 @@ def self.paths=(env)
           target[k] = v
         when Array
           unless Gem::Deprecate.skip
-            warn <<-eowarn
+            warn <<-EOWARN
```
Comment (simi) [inline lib/rubygems.rb:404]: Can we unify this to use `WARNING` as well?

File: test/rubygems/test_gem_ext_cmake_builder.rb [hunk 2]
```diff
@@ -23,11 +23,11 @@ def setup
 
   def test_self_build
     File.open File.join(@ext, 'CMakeLists.txt'), 'w' do |cmakelists|
-      cmakelists.write <<-eo_cmake
+      cmakelists.write <<-EO_CMAKE
```
Comment (simi) [inline test/rubygems/test_gem_ext_cmake_builder.rb:26]: Would `CMAKE` be enough here?

File: test/rubygems/test_gem_remote_fetcher.rb [hunk 3]
```diff
@@ -81,7 +81,7 @@ class TestGemRemoteFetcher < Gem::TestCase
   # Generated via:
   #   x = OpenSSL::PKey::DH.new(2048) # wait a while...
   #   x.to_s => pem
-  TEST_KEY_DH2048 = OpenSSL::PKey::DH.new <<-_end_of_pem_
+  TEST_KEY_DH2048 = OpenSSL::PKey::DH.new <<-_END_OF_PEM_
```
Comment (simi) [inline test/rubygems/test_gem_remote_fetcher.rb:84]: `PEM`?

File: test/rubygems/test_gem_request_set.rb [hunk 4]
```diff
@@ -396,9 +396,9 @@ def test_resolve_git
     rs = Gem::RequestSet.new
 
     tf = Tempfile.open 'gem.deps.rb' do |io|
-      io.puts <<-gems_deps_rb
+      io.puts <<-GEMS_DEPS_RB
```
Comment (simi) [inline test/rubygems/test_gem_request_set.rb:399]: If I understand it well, this is also just code ruby block. Any reason to not use `RUBY` as well?

File: test/rubygems/test_gem_commands_setup_command.rb [hunk 5]
```diff
@@ -359,7 +359,7 @@ def test_show_release_notes
     @cmd.options[:previous_version] = Gem::Version.new '2.0.2'
 
     File.open 'History.txt', 'w' do |io|
-      io.puts <<-History_txt
+      io.puts <<-HISTORY_TXT
```
Comment (simi) [inline test/rubygems/test_gem_commands_setup_command.rb:362]: Would `TEXT` be enough here?

Comment (simi) [reply test/rubygems/test_gem_commands_setup_command.rb:362]: Also can't we use 

```ruby
File.write('History.txt', <<-TEXT)
  ...
TEXT

here?
```

**Comment by simi** [comment]
@deivid-rodriguez I left some minor non-blocking comments from my side.

**Comment by deivid-rodriguez** [comment]
@simi I only made the necessary changes to get the `Naming/HeredocDelimiterCase` cop green since you mentioned it, but I didn't intend to make a whole review of heredoc naming. There's for example a lot of heredocs named `G`, which could be renamed to `GEMFILE` or `RUBY` as well. So much stuff that can be improved in the code base...

Let me get this in and iterate on your suggestions next time I work on a style PR, ok? :)

**Pull request merged**